### PR TITLE
Redux Introduced!

### PR DESCRIPTION
**Pull request by Katedam**
We have a store, reducer, selector, action and in Document.js there's updateState, subscribe and unsubscribe.

### CODE REVIEW

File: src/selectors/selectMarkdown.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const getMarkdown = state => {
+  console.log('state markdown', typeof(state.markdown));
```
Comment (MariahAdams) [inline src/selectors/selectMarkdown.js]:
```suggestion
```
Remember to remove console logs from production code.

File: src/store.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { createStore } from 'redux';
+import reducer from './reducers/reducerDocument';
+
+const store = createStore(
+  reducer,
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
+
```
Comment (MariahAdams) [inline src/store.js]: Clean up whitespace.

**Comment by mehtaphysical** [comment]
Need tests for reducers, actionCreators, and selectors

File: src/components/App.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import Document from './markdown/Document';
 
+
```
Comment (mehtaphysical) [inline src/components/App.js]: what is this space for?

File: src/components/markdown/Document.js [hunk 4]
```diff
@@ -2,16 +2,38 @@ import React, { PureComponent } from 'react';
 import Preview from './Preview';
 import Editor from './Editor';
 import styles from './Document.css';
```
Comment (mehtaphysical) [inline src/components/markdown/Document.js]: I like to import styles as the last item.

File: src/components/markdown/Document.js [hunk 5]
```diff
@@ -2,16 +2,38 @@ import React, { PureComponent } from 'react';
...
 
 export default class Document extends PureComponent {
   state = {
-    markdown: '# Hi there'
+    markdown: '# Hi there',
+    unsubscribe: null
+  };
+
+  updateState = () => {
+    const currentMarkdown = store.getState();
```
Comment (mehtaphysical) [inline src/components/markdown/Document.js]: why call this `currentMarkdown`? Its your entire state, potentially not just markdown

**Review by mehtaphysical — APPROVED** [review]

**Pull request merged**